convert_hf_to_gguf.py

```diff
@@ -217,7 +217,8 @@ def index_tensors(self, remote_hf_model_id: str | None = None) -> dict[str, Call
                     if weight_map is None or not isinstance(weight_map, dict):
                         raise ValueError(f"Can't load 'weight_map' from {index_name!r}")
                     tensor_names_from_index.update(weight_map.keys())
-                    part_names |= set(weight_map.values())
+                    part_dict: dict[str, None] = dict.fromkeys(list(weight_map.values()) + part_names, None)
```

Comment:
```suggestion
                    part_dict: dict[str, None] = dict.fromkeys(weight_map.values(), None)
```
I'm thinking that we should probably treat the index as authoritative and not supplement it with files we've scanned from the model dir.

If this somehow leads to missing tensors because of a bad index old behavior will be restored by deleting the index, WDYT?

Reply:
> I'm thinking that we should probably treat the index as authoritative and not supplement it with files we've scanned from the model dir.

I agree with this. If the weight map is present, and there are somehow more `model*.safetensors` which are scanned, it would cause an error with mismatches anyway.